src/fides/api/models/messaging_template.py

```diff
@@ -56,13 +60,71 @@
 }
 
 
+def _link_properties_to_messaging_template(
+    db: Session,
+    properties: List[Dict[str, Any]],
+    messaging_template: MessagingTemplate,
+) -> Optional[List[Property]]:
+    """
+    Link supplied properties to MessagingTemplate and unlink any properties not supplied.
+    """
+    if len(properties) == 0:
+        return []
+    new_properties = (
+        db.query(Property)
+        .filter(Property.id.in_([prop["id"] for prop in properties]))
+        .all()
+    )
+    messaging_template.properties = new_properties
+    messaging_template.save(db)
+    return messaging_template.properties
+
+
 class MessagingTemplate(Base):
     @declared_attr
     def __tablename__(self) -> str:
         return "messaging_template"
 
-    key = Column(String, index=True, unique=True, nullable=False)
+    type = Column(String, index=True, nullable=False)
     content = Column(MutableDict.as_mutable(JSONB), nullable=False)
+    properties: RelationshipProperty[List[Property]] = relationship(
+        "Property",
+        secondary="messaging_template_to_property",
+        back_populates="messaging_templates",
+        lazy="selectin",
+    )
+    is_enabled = Column(Boolean, default=False, nullable=False)
```

Comment:
Right now, this is what product wants for the UI, where we enable / disable at the template level, not at the property and template level. But I agree, we should probably plan for something more flexible at the db level, just in case. 

I was struggling before with figuring out how to write data to the many-to-many table when we had `is_enabled` there, but I'll see what I can do. 

Maybe some examples you can point me to of other many-to-many tables where we do something similar? I didn't see anything in the SQLAlchemy docs.

Reply:
If that's what product wants, that's fine, enabling/disabling at the template level. I was just double checking.